pennylane/compiler/python_compiler/transforms/convert_to_mbqc_formalism.py

```diff
@@ -0,0 +1,770 @@
+        measureNegOp = MeasureInBasisOp(
+            in_qubit=in_qubit, plane=planeOp, angle=constNegAngleOp.result
+        )
+        # Create the true region
+        true_region = [measureOp, scf.YieldOp(measureOp.results[0], measureOp.results[1])]
+        # Create the false region
+        false_region = [
+            constNegAngleOp,
+            measureNegOp,
+            scf.YieldOp(measureNegOp.results[0], measureNegOp.results[1]),
+        ]
+
+        # Create a If control flow Op
+        condOp = IfOp(
+            cmpOp,
+            (builtin.IntegerType(1), QubitType()),
+            true_region,
+            false_region,
+        )
+        # Insert condOp to the IR
+        rewriter.insert_op(condOp, InsertPoint.before(op))
+
+        # Return the result of if control flow
+        return condOp.results
+
+    def _hadamard_measurements(
+        self, graph_qubits_dict, op, rewriter: pattern_rewriter.PatternRewriter
+    ):
+        """Insert measurement ops for a Hadamard gate and return measurement results and the result graph qubits"""
+        m1, graph_qubits_dict[1] = self._measure_x_op(graph_qubits_dict[1], op, rewriter)
+        m2, graph_qubits_dict[2] = self._measure_y_op(graph_qubits_dict[2], op, rewriter)
+        m3, graph_qubits_dict[3] = self._measure_y_op(graph_qubits_dict[3], op, rewriter)
+        m4, graph_qubits_dict[4] = self._measure_y_op(graph_qubits_dict[4], op, rewriter)
+        return [m1, m2, m3, m4], graph_qubits_dict
+
+    def _s_measurements(self, graph_qubits_dict, op, rewriter: pattern_rewriter.PatternRewriter):
+        """Insert measurement ops for a S gate and return measurement results and the result graph qubits"""
+        m1, graph_qubits_dict[1] = self._measure_x_op(graph_qubits_dict[1], op, rewriter)
+        m2, graph_qubits_dict[2] = self._measure_x_op(graph_qubits_dict[2], op, rewriter)
+        m3, graph_qubits_dict[3] = self._measure_y_op(graph_qubits_dict[3], op, rewriter)
+        m4, graph_qubits_dict[4] = self._measure_x_op(graph_qubits_dict[4], op, rewriter)
+        return [m1, m2, m3, m4], graph_qubits_dict
+
+    def _cnot_measurements(self, graph_qubits_dict, op, rewriter: pattern_rewriter.PatternRewriter):
+        """Insert measurement ops for a CNOT gate and return measurement results and the result graph qubits"""
+        m1, graph_qubits_dict[1] = self._measure_x_op(graph_qubits_dict[1], op, rewriter)
+        m2, graph_qubits_dict[2] = self._measure_y_op(graph_qubits_dict[2], op, rewriter)
+        m3, graph_qubits_dict[3] = self._measure_y_op(graph_qubits_dict[3], op, rewriter)
+        m4, graph_qubits_dict[4] = self._measure_y_op(graph_qubits_dict[4], op, rewriter)
+        m5, graph_qubits_dict[5] = self._measure_y_op(graph_qubits_dict[5], op, rewriter)
+        m6, graph_qubits_dict[6] = self._measure_y_op(graph_qubits_dict[6], op, rewriter)
+        m8, graph_qubits_dict[8] = self._measure_y_op(graph_qubits_dict[8], op, rewriter)
```

Comment:
Been a while since I've looked at the CNOT stencil, but 7 doesn't get measured in this case?